feat(dcgw): Managed cache (#3975)
Kong/developer.konghq.com: app/_how-tos/dedicated-cloud-gateways/aws-managed-cache-control-plane-group.md

1. In the **Name** field, enter `dcgw-control-plane-group`.
1. From the **Control Planes** dropdown menu, select `hybrid-cp`. 
1. For the Node Type, select **Dedicated Cloud**.
1. Click **Save**.
1. Click **Configure data plane**.
1. From the **Provider** dropdown menu, select "AWS".
1. From the **Region** dropdown menu, select the region you want to configure the cluster in. 
1. Edit the Network range as needed.
   
   {:.danger}
   > **Important:** Your AWS virtual network **must** use a different IP than your network in {{site.konnect_short_name}}, which is `10.0.0.0/16` by default but can be edited.
1. From the **Access** dropdown menu, select "Public" or "Private".
1. Click **Create data plane node**.
1. Click **Go to overview**.

   {:.warning}
   > **Important:** Wait until your AWS network displays as `Ready` before proceeding to the next step.

1. Copy and export the control plane group you want to configure the managed cache for:
   ```sh
   export CONTROL_PLANE_GROUP_ID='YOUR CONTROL PLANE GROUP ID'
   ```

### Create a managed cache for your control plane group

{% include_cached /sections/managed-cache-cpg-setup.md %}

## Configure Redis for plugins

> You must manually create a [Redis partial](/gateway/entities/partial/) configuration on each control plane where Redis-backed plugins are enabled. [Use the redis configuration](/gateway/entities/partial/#add-a-partial-to-a-plugin) to setup Redis-supported plugins by selecting the automatically created {{site.konnect_short_name}}-managed Redis configuration. You can’t use the Redis partial configuration in custom plugins. Instead, use env referenceable fields directly.

Reviewer: Copilot is struggling to understand "automatically-managed" here because it's confusing in context. I'd use this edit instead of copilot's:

```suggestion
Next, you must manually create a [Redis partial](/gateway/entities/partial/) configuration on each control plane where Redis-backed plugins are enabled. 
Each Redis partial will automatically use the managed cache that you just set up.

{:.info}
> **Note**: You can’t use the Redis partial configuration in custom plugins. Instead, use [env referenceable fields](/gateway/entities/vault/#store-secrets-as-environment-variables) directly.
```

I replaced this sentence:
> [Use the redis configuration](https://deploy-preview-3975--kongdeveloper.netlify.app/gateway/entities/partial/#add-a-partial-to-a-plugin) to setup Redis-supported plugins by selecting the automatically created Konnect-managed Redis configuration.

I _think_ it's telling me that by creating this partial, it'll automatically use the managed cache. If I misunderstood, feel free to adjust.

url: /v2/control-planes/$CONTROL_PLANE_ID/core-entities/partials
status_code: 201
method: POST
region: us
body:
  name: konnect-managed
  type: redis-ee
  config:
    cloud_authentication:
      auth_provider: "{vault://env/ADDON_MANAGED_CACHE_AUTH_PROVIDER}"
      aws_cache_name: "{vault://env/ADDON_MANAGED_CACHE_AWS_CACHE_NAME}"
      aws_region: "{vault://env/ADDON_MANAGED_CACHE_AWS_REGION}"
      aws_is_serverless: false
      aws_assume_role_arn: "{vault://env/ADDON_MANAGED_CACHE_AWS_ASSUME_ROLE_ARN}"
    connect_timeout: 2000
    connection_is_proxied: false
    database: 0
    host: "{vault://env/ADDON_MANAGED_CACHE_HOST}"
    keepalive_backlog: 512
    keepalive_pool_size: 256
    port: "{vault://env/ADDON_MANAGED_CACHE_PORT}"
    read_timeout: 5000
    send_timeout: 2000
    server_name: "{vault://env/ADDON_MANAGED_CACHE_SERVER_NAME}"
    ssl_verify: true
    ssl: true
    username: "{vault://env/ADDON_MANAGED_CACHE_USERNAME}"
{% endkonnect_api_request %}
<!--vale on-->
{% endcapture %}
{{ create_redis_partial | indent: 3 }}
1. Repeat the previous step for all the control planes in your control plane group.

### Apply the managed cache Redis partial to a plugin

> 

Reviewer: ```suggestion
Let's apply this configuration to a Rate Limiting Advanced plugin.
```
These sections really need intro sentences.

1. In the {{site.konnect_short_name}} sidebar, click **API Gateway**.
1. Click your Dedicated Cloud Gateway.
1. In the API Gateway sidebar, click **Plugins**.
1. Click **New plugin**.
1. Select **Rate Limiting Advanced**.
1. In the **Rate Limit Window Type** fields, enter `100` and `3600`. 
1. Click **View advanced parameters**.
1. In the **Strategy** dropdown menu, select "redis".

> 1. In the **Shared Redis Configuration** dropdown menu, select your {{site.konnect_short_name}}-managed configuration. For example: `konnect-managed`

Reviewer: ```suggestion
1. In the **Shared Redis Configuration** dropdown menu, select your {{site.konnect_short_name}}-managed configuration.
```
I don't think this example adds anything; that would be the same as saying "click your control plane; for example, `hybrid-cp`".

1. In the **Sync Rate** field, enter `5`.
1. Click **Save**.
1. Repeat steps 1 - 11 for each control plane in your control plane group.
   

>    You can add more data plane groups to the control plane group or remove existing data plane groups. The cache is automatically updated accordingly. You can check the managed cache's ready status to make sure the managed cache is up-to-date.

Reviewer: This feels like it belongs in an FAQ, eg "will new data planes/data plane groups use an existing managed cache?". Right now the info seems out of place here.

Author: We mention that the CPs all share the managed cache, so I don't think this is adding anything. I'll remove it.
